retriever/lib/scripts.py

```diff
@@ -107,7 +108,101 @@ def get_script(dataset):
     if dataset in scripts:
         return scripts[dataset]
     else:
-        raise KeyError("No dataset named: {}".format(dataset))
+        if dataset in RETRIEVER_DATASETS:
+            read_script = get_script_upstream(dataset, repo=RETRIEVER_REPOSITORY)
+        else:
+            read_script = get_script_upstream(dataset)
+        if read_script is None:
+            raise KeyError("No dataset named: {}".format(dataset))
+        else:
+            return read_script
```

Comment:
First checks if the script is available locally, if present returns it. Then, checks if it is a `retriever legacy` script or not. Then downloads it from the according upstream repository.